### DISCUSSION TITLE
Sample Client

### DISCUSSION THREAD

**Discussion by marcogruhl** [Xwilarg/TwitterSharp#37] [Ideas]
I would write a small WPF client if you think this is a good idea ...

**Comment by Xwilarg** [comment]
That sounds like a great idea, would give a good example on how to use the library
However I wonder if WPF is the best choice since it's Windows only

**Reply by marcogruhl** [reply]
WPF is what i'm familiar with. What are you thinking of? MAUI? ASP? Blazor? Console?

**Reply by Xwilarg** [reply]
Ah if you're more familliar with WPF it's okay

On my side I'm usually using Avalonia which is crossplatform

**Reply by marcogruhl** [reply]
I think I go with WPF ... later we can add other clients!? That looks fun: [Terminal.Gui](https://github.com/migueldeicaza/gui.cs) 😄
PS: Avalonia looks very interesting ... almost exactly like WPF.

**Reply by Xwilarg** [reply]
It's the first time I heard of Terminal.Gui, it indeed looks really cool

**Comment by marcogruhl** [comment]
The Sample Client is completed.

My goals were:

- visualize the way the API works (Expression -> Rule -> Stream)
- implement all tweet related functionality
- fill in working default values
- persist configuration
- don't use external libraries

Future development

- reset expression to default values
- load expression from rule
- more clean up
- all non tweet related functionality
- console client
- more non-api stuff (style, sorting, ...)

It's gotten a little bigger then I intentionally thought 😅

I am not sure if i should create a pull request or create a complete new repository.

Please look at it share your thoughts about it! You can find it here: [WPF Client](https://github.com/marcogruhl/TwitterSharp/tree/WPFClientAdvanced)

Screenshot:
![T#Client](https://user-images.githubusercontent.com/9695930/183319747-3da5d5f1-6230-44f2-a5d1-d83e3375daaa.png)

**Reply by Xwilarg** [reply]
Ah I don't think I ever saw that message, oops
But your client looks really great! Want to add it to the repository?

**Reply by marcogruhl** [reply]
No Problem.
Since then i put it in an own repository ([TweetCommander](https://github.com/marcogruhl/TweetCommander)) to take part on the [Chirp Developer Challenge](https://devpost.com/software/twittercommander).
Currently i tend to make a simple console client with the samples from the README and maybe some more to cover the whole (implemented) functionality of the api.
like i said, the "TweetCommander" is maybe a little overblown as sample for an api.
We can put a link to it as an example for a visual implementation ...

**Reply by Xwilarg** [reply]
That's a pretty cool website, I didn't know about it and sure, we can put it in the README I think it could be a great idea
